lib/src/browser/disposable_browser.dart

```diff
@@ -203,17 +204,17 @@ class Disposable implements disposable_common.Disposable {
     _disposable
       ..onDisposeHandler = onDispose
       ..onWillDisposeHandler = onWillDispose;
-    return _disposable.dispose().then((_) {
-      // We want the description to be the runtime type of this
-      // object, not the proxy disposable, so we need to reset
-      // the leak flag here.
-      flagLeak(runtimeType.toString());
-    });
+    // We want the description to be the runtime type of this
+    // object, not the proxy disposable, so we need to set
+    // the leak flag here, before we delegate the `dispose`
+    // call.
```

Comment:
Should we call out that this may cause false positive leak flags when the object takes a while to dispose?

Reply:
Technically that's always possible since the actual process of disposal is unrelated to whatever process actually causes the object to become available for GC, so I would argue that this doesn't really change anything for consumers.

Reply:
> the actual process of disposal is unrelated to whatever process actually causes the object to become available for GC

I would disagree with that; I thought most disposed weren't meant to be interacted with any further, and should be made available to be GC'd ASAP.